proxy/src/auth/backend.rs

```diff
@@ -139,14 +145,38 @@ async fn auth_quirks_creds(
     // If there's no project so far, that entails that client doesn't
     // support SNI or other means of passing the endpoint (project) name.
     // We now expect to see a very specific payload in the place of password.
-    if creds.project.is_none() {
+    let maybe_success = if creds.project.is_none() {
         // Password will be checked by the compute node later.
-        return hacks::password_hack(creds, client, latency_timer).await;
-    }
+        Some(hacks::password_hack(creds, client, latency_timer).await?)
+    } else {
+        None
+    };
 
     // Password hack should set the project name.
     // TODO: make `creds.project` more type-safe.
     assert!(creds.project.is_some());
+    info!("fetching user's authentication info");
+    // TODO(anna): this will slow down both "hacks" below; we probably need a cache.
+    let AuthInfo {
+        secret,
+        allowed_ips,
+    } = api.get_auth_info(extra, creds).await?;
```

Comment:
The slowdown should be minimal once the cp split is done. I also think it's good to keep this consistent as our users shouldn't choose password hack because "it's faster".

However, websocket's use the cleartext hack and will be slowed down - but this is already fairly high latency cost.
My biggest concern is HTTP auth (in another file) which we **need** to be fast (<10ms)

Reply:
Ok, I see. Ideally, we also should separate secret and allowed ips in two different methods. I think, for the hacks it's ok to keep it as it is.